server/src/main/java/org/elasticsearch/search/query/QueryPhaseCollector.java

```diff
@@ -389,77 +374,6 @@ boolean incrementHitCountAndCheckThreshold() {
         }
     };
 
-    /**
-     * {@link org.apache.lucene.search.CollectorManager} implementation based on {@link QueryPhaseCollector}.
-     * Wraps two {@link org.apache.lucene.search.CollectorManager}s: one required for top docs collection, and another one optional for
-     * aggs collection. Applies terminate_after consistently across the different collectors by sharing an atomic counter of collected docs.
-     */
-    static class CollectorManager implements org.apache.lucene.search.CollectorManager<QueryPhaseCollector, Void> {
```

Comment:
This functionality is now part of QueryPhaseCollectorManager. This collector manager has been simplified and moved to QueryPhaseCollectorTests to allow for testing QueryPhaseCollector in concurrent scenarios, as opposed to testing it only in sequential scenarios.